Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialVersionUpdate.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RPI.Reflect/Material/MaterialVersionUpdate.h>
+#include <Atom/RPI.Reflect/Material/MaterialAsset.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        void MaterialVersionUpdate::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                serializeContext->Class<MaterialVersionUpdate::Action>()
+                    ->Version(1)
+                    ->Field("ArgsMap", &Action::m_argsMap)
+                    ->Field("Operation", &Action::m_operation)
+                    ;
+
+                serializeContext->RegisterGenericType<MaterialVersionUpdate::Actions>();
+
+                serializeContext->Class<MaterialVersionUpdate>()
+                    ->Version(1)
+                    ->Field("ToVersion", &MaterialVersionUpdate::m_toVersion)
+                    ->Field("Actions", &MaterialVersionUpdate::m_actions)
+                    ;
+
+                serializeContext->RegisterGenericType<MaterialVersionUpdateMap>();
+            }
+        }
+
+        MaterialVersionUpdate::MaterialVersionUpdate(uint32_t toVersion)
+            : m_toVersion(toVersion)
+        {
+        }
+
+        uint32_t MaterialVersionUpdate::GetVersion() const
+        {
+            return m_toVersion;
+        }
+
+        void MaterialVersionUpdate::SetVersion(uint32_t toVersion)
+        {
+            m_toVersion = toVersion;
+        }
+
+        void MaterialVersionUpdate::ApplyVersionUpdates(MaterialAsset& materialAsset) const
+        {
+            // collect all renames within this version update
+            AZStd::unordered_map<AZ::Name, AZ::Name> renameToFrom;
+            for (const auto& action : m_actions)
+            {
+                const AZ::Name from = action.m_argsMap.find(AZ::Name{ "from" })->second;
+                const AZ::Name to = action.m_argsMap.find(AZ::Name{ "to" })->second;
```

Comment:
We should at least move these names out of the loop so they are only created once.

Reply:
It would be even better to make sure they are only created once per MaterialVersionUpdate object. 

(Actually it would be best to make them created only once, maybe in the MaterialSystem class, but let's save that until we have data ... I think that would be premature optimization).